index.js

```diff
@@ -466,32 +500,41 @@ Client.prototype._isUnsafeToWrite = function () {
   return this.destroyed
 }
 
+Client.prototype._shouldDropEvent = function () {
```

Comment:
Do we want some sort of logging that indicates the client or agent is dropping events?

Reply:
From [earlier chat](https://elastic.slack.com/archives/C5M02D1GE/p1615912120010100):

One last thing I'm considering here is changing the agent to use process.emitWarning when we want to give warnings to the user rather than logger.warn() or logger.error(). That will:
1. allow us to not spam to user if it is a repeated situation (e.g. if we are dropping events because of load on the APM server)
2. make the text more readable for the user (it won't be a JSON log record)
3. allow the user to handle/silence particular warnings if they want

Sound reasonable if I defer that to a separate ticket?

Reply:
https://github.com/elastic/apm-agent-nodejs/issues/2029 added to track this. Thanks for bringing it up.

Reply:
> Sound reasonable if I defer that to a separate ticket?

👍